web/src/user_profile.js

```diff
@@ -702,6 +704,18 @@ export function show_edit_user_info_modal(user_id, $container) {
         user_deactivation_ui.confirm_deactivation(user_id, handle_confirm, true);
     });
 
+    // Handle reactivation
+    $("#edit-user-form").on("click", ".reactivate_user_button", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const user_id = $("#edit-user-form").data("user-id");
+        function handle_confirm() {
+            const url = "/json/users/" + encodeURIComponent(user_id) + "/reactivate";
+            dialog_widget.submit_api_request(channel.post, url);
+        }
+        settings_users.confirm_reactivation(user_id, handle_confirm, true);
+    });
```

Comment:
While testing this, I noticed we need parallel code for the bot case:

``` diff
diff --git a/web/src/user_profile.js b/web/src/user_profile.js
index 71f275ffda..407ed9532d 100644
--- a/web/src/user_profile.js
+++ b/web/src/user_profile.js
@@ -440,6 +440,7 @@ export function show_edit_bot_info_modal(user_id, $container) {
     const bot = people.maybe_get_user_by_id(user_id);
     const owner_id = bot_data.get(user_id).owner_id;
     const owner_full_name = people.get_full_name(owner_id);
+    const is_active = people.is_person_active(user_id);
 
     if (!bot || !bot_data.get(user_id)) {
         return;
@@ -447,6 +448,7 @@ export function show_edit_bot_info_modal(user_id, $container) {
 
     const html_body = render_edit_bot_form({
         user_id,
+        is_active,
         email: bot.email,
         full_name: bot.full_name,
         user_role_values: settings_config.user_role_values,
@@ -604,6 +606,18 @@ export function show_edit_bot_info_modal(user_id, $container) {
             user_deactivation_ui.confirm_bot_deactivation(bot_id, handle_confirm, true);
         });
 
+        // Handle reactivation
+        $("#bot-edit-form").on("click", ".reactivate_user_button", (e) => {
+            e.preventDefault();
+            e.stopPropagation();
+            const user_id = $("#bot-edit-form").data("user-id");
+            function handle_confirm() {
+                const url = "/json/users/" + encodeURIComponent(user_id) + "/reactivate";
+                dialog_widget.submit_api_request(channel.post, url);
+            }
+            user_deactivation_ui.confirm_reactivation(user_id, handle_confirm, true);
+        });
+
         $("#bot-edit-form").on("click", ".generate_url_for_integration", (e) => {
             e.preventDefault();
             e.stopPropagation();
diff --git a/web/templates/settings/edit_bot_form.hbs b/web/templates/settings/edit_bot_form.hbs
index 1e5fd0041c..3878138604 100644
--- a/web/templates/settings/edit_bot_form.hbs
+++ b/web/templates/settings/edit_bot_form.hbs
@@ -50,8 +50,16 @@
     </div>
     {{/if}}
     <div class="input-group new-style">
+        {{#if is_active}}
         <button class="button rounded btn-danger deactivate_bot_button">
             {{t 'Deactivate bot' }}
         </button>
+        {{else}}
+        <span>
+            <button class="button rounded reactivate_user_button">
+                {{t 'Reactivate user' }}
+            </button>
+        </span>
+        {{/if}}
     </div>
 </div>
```

It would probably be a useful follow-up to try to reduce duplication between the bot/user code paths for details like this where the difference is not important.